build-tools/packages/build-cli/src/commands/generate/changelog.ts

```diff
@@ -84,24 +85,40 @@ export default class GenerateChangeLogCommand extends BaseCommand<
 	}
 
 	/**
-	 * Removes any additional metadata from all changesets and writes the output back to the source files.
+	 * Removes any custom metadata from all changesets and writes the output back to the source files. Changesets that
```

Comment:
What is "custom metadata"?

There are a lot of things that could mean (ex: some file systems support adding custom tags to files, it could be referring to those, or it could be some format for encoding metadata that is custom meaning that its user defined, and this tooling does not understand it (like sections in addition to those our changeset generator creates)

I'm guessing this is referring to the header section of the changeset files. If that is accurate I'd call this something like: stripNonMarkDownHeader to indicate this removes the header, and the resulting file is actual proper markdown.

Reply:
I now see your response in the other comment thread. Generally if you need to explain a doc comment, you should instead modify the doc comment to contain that explanation. In this case, maybe link https://github.com/microsoft/FluidFramework/wiki/Changesets#custom-metadata , and note as you did in the othert thread that this modifies the changesets to be compatible with  some third party tool (which should be explicitly named here so readers can look up what format it expects)

Reply:
The full doc comment now reads:

> Removes any custom metadata from all changesets and writes the resulting changes back to the source files. This
> metadata needs to be removed prior to running `changeset version` from the \@changesets/cli package. If it is not,
> then the custom metadata is interpreted as part of the content and the changelogs end up with the metadata in them.
> 
> For more information about the custom metadata we use in our changesets, see
> https://github.com/microsoft/FluidFramework/wiki/Changesets#custom-metadata
>
> **Note that this is a lossy action!** The metadata is completely removed. Changesets are typically in source
> control so changes can usually be reverted.

In addition, the wiki now explcitly calls out multiple times that the custom metadata is not compatible with tools like \@changesets/cli using callouts like the following:

> [!IMPORTANT]
> **Our custom metadata is not supported by standard changeset tools, including [@changesets/cli](https://www.npmjs.com/package/@changesets/cli),** which is the core changesets package.
> Because of this, some of our tools such as `flub generate:changelog` strip the changesets of this metadata before invoking commands like `changeset version`.
> If the custom metadata is not removed, then it ends up in the changelogs, which messes with the changelog formatting and is also confusing.